### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by shardulc**
A lot of tests have been modified, and a few modules wherever I thought the problem was in the module, not the test. For `test_head.py`, I mock all web requests because [`head.py`](https://github.com/goavki/phenny/blob/master/modules/head.py)'s core functionality is distinct from the success of web requests; for other modules like [`fml.py`](https://github.com/goavki/phenny/blob/master/modules/mylife.py) which have the web service as the primary component, the tests check whether that service is available and decide to skip or proceed. (These might all eventually need to be mocked.) Other changes include replacing bare `assert`s with the `unittest` versions, formatting, and small module fixes.

### CODE REVIEW

**Comment by shardulc** [comment]
Right now, builds with Python 3.5, 3.6, and dev pass on Travis, so #199 should be fixed.

**Comment by wei2912** [comment]
Looks like only Python 3.5 builds. Could you do the following:
1) Look into why it fails for other versions of Python.
2) Look through warnings on https://travis-ci.org/goavki/phenny/jobs/187803700 and fix them, so that our logs won't be so cluttered.

**Comment by shardulc** [comment]
Python 3.2 fails while installing `nose` because of newer syntax.
Python 3.3 fails because some tests use newer functions in the `re` (regex) module.
All newer Python builds pass.

**Comment by wei2912** [comment]
Can you disable Python 3.2 and 3.3 in .travis.yml? I'll approve your task.

File: modules/archwiki.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 import web
 import wiki
 
-wikiapi = 'https://wiki.archlinux.org/api.php?action=query&list=search&srsearch={0}&limit=1&prop=snippet&format=json'
```
Comment (wei2912) [inline modules/archwiki.py]: How does this help?

File: modules/test/test_clock.py [hunk 2]
```diff
@@ -2,75 +2,67 @@
...
 
 
+@patch('time.time')
 class TestClock(unittest.TestCase):
     def setUp(self):
         self.phenny = MagicMock()
+        self.input = MagicMock()
 
-    @patch('time.time')
     @nottest
```
Comment (wei2912) [inline modules/test/test_clock.py]: Should be disabled so that we can test if this test works.

File: modules/archwiki.py (hunk 1 above)
Comment (shardulc) [inline modules/archwiki.py]: The `prop=snippet` parameter is not supported by the ArchWiki API [reference](https://wiki.archlinux.org/api.php) and was causing the module to work incorrectly; without it, snippets are still available in the search results returned.

File: modules/test/test_clock.py (hunk 2 above)
Comment (shardulc) [inline modules/test/test_clock.py]: So should I disable it or enable it to see if it works?

**Comment by wei2912** [comment]
Thanks a lot for this extensive set of changes!

**Pull request merged**